rocketpy/Rocket.py

```diff
@@ -537,56 +528,133 @@ def addFins(
         self : Rocket
             Object of the Rocket class.
         """
-
-        # Retrieve parameters for calculations
-        Cr = rootChord
-        Ct = tipChord
-        Yr = rootChord + tipChord
-        s = span
-        Af = Yr * s / 2  # fin area
-        Yma = (
-            (s / 3) * (Cr + 2 * Ct) / Yr
-        )  # span wise position of fin's mean aerodynamic chord
-        gamac = np.arctan((Cr - Ct) / (2 * s))
-        Lf = np.sqrt((Cr / 2 - Ct / 2) ** 2 + s**2)
+        # Checking the dictionary
+        if "rootChord" not in finParameters.keys():
+            raise Exception(
+                "rootChord dictionary key is missing from finParameters")
+        if "span" not in finParameters.keys():
+            raise Exception(
+                "span dictionary key is missing from finParameters")
+
+        if type == "trapezoidal":
+            if "tipChord" not in finParameters.keys():
+                raise Exception(
+                    "tipChord dictionary key is missing from finParameters")
+
+        # Retrieves similar parameters
+        Cr = finParameters["rootChord"]
+        s = finParameters["span"]
         radius = self.radius if radius == 0 else radius
+        cantAngleRad = np.radians(cantAngle)
         d = 2 * radius
+
+        # Type verification
+        if type == "trapezoidal":
+
+            # Retrieve parameters for calculations
+            Ct = finParameters["tipChord"]
+            Yr = Cr + Ct
+
+            Af = Yr * s / 2  # fin area
+            gamac = np.arctan((Cr - Ct) / (2 * s))  # mid chord angle
+            Ymac = (
+                (s / 3) * (Cr + 2 * Ct) / Yr
+            )  # span wise position of fin's mean aerodynamic chord
+            Lf = np.sqrt((Cr / 2 - Ct / 2) ** 2 + s**2)  #
+
+            rollGeometricalConstant = (
+                (Cr + 3 * Ct) * s**3
+                + 4 * (Cr + 2 * Ct) * radius * s**2
+                + 6 * (Cr + Ct) * s * radius**2
+            ) / 12
+
+            # Calculate cp position relative to cm
+            if distanceToCM < 0:
+                cpz = distanceToCM - (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+            else:
+                cpz = distanceToCM + (
+                    ((Cr - Ct) / 3) * ((Cr + 2 * Ct) / (Cr + Ct))
+                    + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
+                )
+
+        elif type == "elliptical":
+
+            # Retrieve parameters for calculations
+            Af = (np.pi * Cr / 2 * s) / 2
+            gamac = 0
+            Ymac = s / (3 * np.pi) * np.sqrt(9 * np.pi**2 - 64)
+            rollGeometricalConstant = (
```

Comment:
The `rollGeometricalConstant`is currently not being used in any other part of the code.
It's probably a misunderstanding with the name definition of this variable.

Please start using it or comment this line